src/Ocelot/Configuration/File/FileQoSOptions.cs

```diff
@@ -32,6 +32,6 @@ public FileQoSOptions(QoSOptions from)
 
         public int DurationOfBreak { get; set; }
         public int ExceptionsAllowedBeforeBreaking { get; set; }
-        public int TimeoutValue { get; set; }
+        public int? TimeoutValue { get; set; }
```

Comment:
We might see some side effects in my opinion. Since, i'm not sure, but we are comparing TimeoutValue with zero somewhere.

Reply:
@hogwartsdeveloper Propagate nullable everywhere up to end consumer, our provider.
Inside of provider add null check and substitute default value for QoS provider only.